src/Core/Services/MetadataProviders/MsSqlMetadataProvider.cs

```diff
@@ -236,5 +240,24 @@ private bool TryResolveDbType(string sqlDbTypeName, out DbType dbType)
                 return false;
             }
         }
+
+        /// <summary>
+        /// This method returns whether the given value is a built_in function or not.
+        /// built_in funtions are surrounded by single parenthesis,i.e. (getdate())
```

Comment:
nit: Please clarify why you are considering these three categories. 1 reason I can think of is "given value could be in one of these 3 formats: 1) when it is a built-in function 2) when it is a string 3) when it is of type int/float". 
As a followup, what happens if the data type is DateTime with a specific default value?

Reply:
if DateTime field has a specific default value, it won't cause any problem since it would be parsed into DateTime.

Reply:
It may not be causing any parsing problems, but what is the format in which the default value is generated? Is it similar to int/float e.g.((01-04-2024)) ? or is it like a string ('01-04-2024') ? How about other scalar types than string? 
Basically my question, are we sure the default value format regardless of any other underlying type will be 1 of these 3 formats?

Reply:
it will be like string. added a test for the same. Thanks for bringing up this case.

Reply:
it will be like string ('01-04-2024'), added a test as well

Reply:
please add it to the comment summary section as well!